### PR TITLE
feat: added retry and timeout params to partition read in database and snapshot class

### LINKED ISSUE #71: Add support for custom timeout and retry parameters through object surfaces
As noted in #67, for most methods there is no way to set timeout and retry parameters without using the GAPIC client directly. 

Adding support for custom timeout and retries will give users more flexibility if they require it.

### PR DESCRIPTION
**Pull request by vi3k6i5**
Added retry and timeout params to partition read in database and snapshot class

Fixes #71 🦕

### CODE REVIEW

**Review by larkee — COMMENTED** [review]
LGTM so far :+1:

`generate_read_batches` and `generate_query_batches` both generate partitions which are then used with `process_read_batch` and `process_query_batch` respectively. Please also add the options to these methods to complete the support

**Comment by vi3k6i5** [comment]
@larkee : made the change.

**Review by larkee — COMMENTED** [review]
LGTM other than some minor test nits

File: tests/unit/test_snapshot.py [hunk 1]
```diff
@@ -926,6 +991,17 @@ def test_partition_query_ok_w_size(self):
     def test_partition_query_ok_w_max_partitions(self):
         self._partition_query_helper(multi_use=True, w_txn=True, max_partitions=4)
 
+    def test_partition_query_ok_w_timeout_param(self):
+        self._partition_query_helper(multi_use=True, w_txn=True, timeout=2.0)
+
+    def test_partition_query_ok_w_retry_param(self):
+        self._partition_query_helper(multi_use=True, w_txn=True, retry={})
```
[the remark this answers is not in the document]
Comment (vi3k6i5) [reply tests/unit/test_snapshot.py]: Changed.

File: tests/unit/test_snapshot.py [hunk 2]
```diff
@@ -926,6 +991,17 @@ def test_partition_query_ok_w_size(self):
...
+
+    def test_partition_query_ok_w_timeout_and_retry_params(self):
+        self._partition_query_helper(
+            multi_use=True, w_txn=True, retry=gapic_v1.method.DEFAULT, timeout=2.0
```
[the remark this answers is not in the document]
Comment (vi3k6i5) [reply tests/unit/test_snapshot.py]: Done.

**Review by larkee — APPROVED** [review]
LGTM :)

**Pull request merged**